grails-app/controllers/org/bbop/apollo/SequenceController.groovy

```diff
@@ -213,20 +228,153 @@ class SequenceController {
+        render view: "report", model: [sequenceInstanceList: sequenceInstanceList, organism: organism, sequenceInstanceCount: sequenceInstanceCount]
+    }
+
+    @RestApiMethod(description = "Get sequence data within a range", path = "/sequence/<organism name>/<sequence name>:<fmin>..<fmax>?ignoreCache=<ignoreCache>", verb = RestApiVerb.GET)
+    @RestApiParams(params = [
+            @RestApiParam(name = "organismString", type = "string", paramType = RestApiParamType.QUERY, description = "Organism common name or ID(required)")
+            , @RestApiParam(name = "sequenceName", type = "string", paramType = RestApiParamType.QUERY, description = "Sequence name(required)")
+            , @RestApiParam(name = "fmin", type = "integer", paramType = RestApiParamType.QUERY, description = "Minimum range(required)")
+            , @RestApiParam(name = "fmax", type = "integer", paramType = RestApiParamType.QUERY, description = "Maximum range (required)")
+            , @RestApiParam(name = "ignoreCache", type = "boolean", paramType = RestApiParamType.QUERY, description = "(default false).  Use cache for request if available.")
+    ])
+    @Transactional
+    String sequenceByLocation(String organismString, String sequenceName, int fmin, int fmax) {
+
+        Boolean ignoreCache = params.ignoreCache != null ? Boolean.valueOf(params.ignoreCache) : false
+        Map paramMap = new TreeMap<>()
+
+        if (!ignoreCache) {
+            String responseString = sequenceService.checkCache(organismString, sequenceName, fmin, fmax,  paramMap)
+            if (responseString) {
+                render responseString
+                return
+            }
+        }
+
+        Organism organism = Organism.findByCommonName(organismString) ?: Organism.findById(organismString as Long)
+        Sequence sequence = Sequence.findByNameAndOrganism(sequenceName, organism)
+
+        Strand strand = params.strand ? Strand.getStrandForValue(params.strand as Integer) : Strand.POSITIVE
+        String sequenceString = sequenceService.getGenomicResiduesFromSequenceWithAlterations(sequence, fmin, fmax, strand)
+        sequenceService.cacheRequest(sequenceString, organismString, sequenceName, fmin, fmax,  paramMap)
+        render sequenceString
+
     }
 
+    @RestApiMethod(description = "Get sequence data as for a selected name", path = "/sequence/<organism name>/<sequence name>/<feature name>.<type>?ignoreCache=<ignoreCache>", verb = RestApiVerb.GET)
+    @RestApiParams(params = [
+            @RestApiParam(name = "organismString", type = "string", paramType = RestApiParamType.QUERY, description = "Organism common name or ID(required)")
+            , @RestApiParam(name = "sequenceName", type = "string", paramType = RestApiParamType.QUERY, description = "Sequence name(required)")
+            , @RestApiParam(name = "featureName", type = "string", paramType = RestApiParamType.QUERY, description = "If top-level feature 'id' matches, then annotate with 'selected'=1")
```

Comment:
Its ambiguous if its a feature `name` or feature `uniquename`. 
Wouldn't it make sense to say `featureUniqueName` or `uniqueName` instead?

Reply:
That is because it can be either.  I'll update the doc to represent that.